backend/src/packages/users/user.repository.ts

```diff
@@ -106,15 +111,41 @@ class UserRepository implements Repository {
     return Promise.resolve(null);
   }
 
-  public delete(): ReturnType<Repository['delete']> {
-    return Promise.resolve(true);
+  public async delete(id: number): Promise<UserEntity | null> {
+    const user = await this.userModel
+
+      .query()
+      .modify('withoutPassword')
+      .withGraphJoined(UsersRelation.DETAILS)
+      .findById(id)
+      .castTo<UserCommonQueryResponse | undefined>()
+      .execute();
+
+    if (!user) {
+      return null;
+    }
+
+    await this.userModel.query().patch({ deletedAt: new Date() }).where({ id });
+
+    user.deletedAt = new Date().toISOString();
+
+    return UserEntity.initialize({
+      id: user.id,
+      email: user.email,
+      createdAt: new Date(user.createdAt),
+      updatedAt: new Date(user.updatedAt),
+      fullName: user.details?.fullName ?? '',
+      isSurveyCompleted: user.details?.isSurveyCompleted ?? false,
+      deletedAt: user.deletedAt ? new Date(user.deletedAt) : null,
+    });
```

Comment:
we can just return boolean instead of this
```suggestion
    return true
```